database/dbHelpers.js

```diff
@@ -39,12 +39,79 @@ const insertFollow = (follow, done) => {
   client.create(record, (err, resp) => {
     if(err) { console.log(err); }
     else {
-      console.log('record inserted successfully, record:', resp);
+      console.log('record inserted successfully, record: ', resp);
       done();
+      trans.end();      
+    }
+  });
+}
+
+const getFeed = (user, done, trans) => {
+  let { user_id } = user;
+  client.search({
+    index: 'follows',
+    q: 'follower_id:' + user_id
+  }, (err, resp) => {
+    if(err) { console.log(err); }
+    else {
+      sortFollows(resp.hits.hits, done, trans);
     }
   })
 }
 
+const sortFollows = (follows, done, trans) => {
+  follows.sort((a, b) => {
+    let aEngagement = parseFloat(a._source.engagement_rating);
+    let bEngagement = parseFloat(b._source.engagement_rating);
+
+    if(aEngagement - bEngagement > 0) {
+      return -1;
+    } else if (aEngagement - bEngagement < 0) {
+      return 1;
+    }
+  });
+  getFeedAuthors(follows, done, trans);
+};
+
+const getFeedAuthors = (follows, done, trans) => {
+  let top5FollowRelationships = follows.slice(0, 5);
+  let usersToBeInFeed = top5FollowRelationships.map((follow) => follow._source.followed_id);
+  getTweetsForFeed(usersToBeInFeed, done, trans);
+};
+
+const getTweetsForFeed = (users, done, trans) => {
+  let userIds = users.map(user=> user).join(' OR ');
+  client.search({
+    index: 'slimtweets',
+    type: 'tweet',
+    q: 'user_id: ' + userIds,
+    size: 10000
```

Comment:
In some ORMs, you can specify the sort order when it comes back from the db. It would be interesting to test the numbers on sorting in the db query as opposed to sorting afterward.